tests/potential/test_potential.py

```diff
@@ -581,5 +577,195 @@ def test_saveload(self):
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2.5, rmin=0.5, rmax=1.5)
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2, rmin=1.0, rmax=0.5)
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2, rmin=-1.0, rmax=1.0)
+
+    def test_potential(self):
+        """Test energy and force methods"""
+        p1 = QuadPot(types=('1',), params=('m',), default={'m':2})
+        p2 = QuadPot(types=('1','2'), params=('m',), default={'m':1})
+        p_all = [p1, p2]
+
+        #test energy method
+        #test with edges=True
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0)
+        u = t.energy(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([0,3,12,27,48]))
+        u = t.energy(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([0,1,4,9,16]))
+        #test with edges=False
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0, edges=False)
+        u = t.energy(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([0.75,6.75,18.75,36.75]))
+        u = t.energy(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(u, np.array([0.25,2.25,6.25,12.25]))
+
+        #test force method
+        #test with edges=True
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0)
+        f = t.force(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([0,-6,-12,-18,-24]))
+        f = t.force(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([0,-2,-4,-6,-8]))
+        #test with edges=False
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0, edges=False)
+        f = t.force(pair=('1','1'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([-3,-9,-15,-21]))
+        f = t.force(pair=('1','2'), potentials=p_all)
+        np.testing.assert_allclose(f, np.array([-1,-3,-5,-7]))
+
+    def test_regularize(self):
+        """Test regularize_force method cases"""
+        p1 = QuadPot(types=('1',), params=('m',), default={'m':2})
+        p2 = QuadPot(types=('1','2'), params=('m',), default={'m':1})
+        p_all = [p1, p2]
+
+        #test without fmax or fcut
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0)
+        u = t.energy(pair=('1','1'), potentials=p_all)
+        f = t.force(pair=('1','1'), potentials=p_all)
+        rcut,stack = t.regularize_force(u, f)
+        self.assertEqual(rcut, None)
+        np.testing.assert_allclose(stack, np.array([[1, 0,  0],
+                                                    [2, 3, -6],
+                                                    [3,12,-12],
+                                                    [4,27,-18],
+                                                    [5,48,-24]]))
+        #test with only fmax
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0, fmax=-10)
```

Comment:
I sort of noted this above, but we will want to make `fmax` and `fcut` >= 0. Can you modify these tests accordingly, and also add an error check for this in `__init__`?